quantumCoinFlipping/1_Honest_Participants/aliceCoin.py

```diff
@@ -0,0 +1,82 @@
+from SimulaQron.general.hostConfig import *
+from SimulaQron.cqc.backend.cqcHeader import *
+from SimulaQron.cqc.pythonLib.cqc import *
+
+import random
+import numpy
+import math
+#########################################
+#
+# FUNCTIONS
+#
+def ACKSend(name, target, data):
+    name.sendClassical(target, data)
+    while bytes(name.recvClassical(msg_size=3)) != b'ACK':
+        pass
+
+def ACKRecv(name, target):
+    data = list(name.recvClassical())
+    name.sendClassical(target, list(b'ACK'))
+    return data
+
+def addNoise(q, noise):
+    if random.random()<noise:
+        q.X()
+    if random.random()<noise:
+        q.Z()
+    return q
+
+def prepareQubit(name, alpha, c, y, noise):
+    q=qubit(name)
+    if c==0:
+        q.rot_Y(round((256/math.pi)*math.acos(math.sqrt(y))))
+        q.rot_Z(128*alpha)
+    elif c==1:
+        q.rot_Y(round((256/math.pi)*math.acos(math.sqrt(1-y))))
+        q.rot_Z(128*((alpha+1)%2))
+    q=addNoise(q, noise)
+    return q
+
+def QcoinFlipAlice(name, target, aliceString, y, K, noise):
+    for k in range (0,K):
+        q=prepareQubit(name, aliceString['alpha'][k], aliceString['c'][k], y, noise)
+        name.sendQubit(q, target)
+        ACKRecv(name, target)
+
+#########################################
+#
+# MAIN
+#
+def main():
+    # Get parameters
+    config=numpy.loadtxt("config.txt")
+    K=int(config[0])
+    y=float(config[1])
+    noise=float(config[3])
+    
+    # Create strings: alpha_i in {0,1} and c_i in {0,1}
+    aliceString={'alpha':[],'c':[]}
+    for k in range (0,K):
+        aliceString['alpha'].append(random.randint(0,1))
+        aliceString['c'].append(random.randint(0,1))
+    # Initialise connection
+    Alice=CQCConnection("Alice")
```

Comment:
`with CQCConnection("Alice") as Alice: 
[...]
` is the recommanded way of doing it